src/hw/maple/maple.c

```diff
@@ -19,12 +19,15 @@ static bool maple_init(struct device *dev) {
   return true;
 }
 
-static void maple_keydown(struct device *d, enum keycode key, int16_t value) {
+static void maple_keydown(struct device *d, enum keycode key, int16_t value, uint8_t index) {
   struct maple *mp = (struct maple *)d;
+  struct maple_port *port = &mp->ports[index];
```

Comment:
Oh I see, you prevent dereferencing the pointer below with some if guards.

I think early outing before ever defining the pointer would be less error prone with regards to future maintenance, and cut down on some of the subsequent conditionals, e.g.:

`if (index >= MAPLE_NUM_PORTS) {
  return;
}`